File: leetcode-algorithms/001. Two Sum/twoSum.js
/*
this function is too slow. We need to have another solution.
*/

function twoSum(nums, target){
	var tmp,
		position,
		a = [];
	for(var i = 0; i < nums.length; i++){
		tmp = target - nums[i];
		position = nums.indexOf(tmp);
		if(position !== i && position !== -1){
		    a.push(i, position);
		    break;
		}
	}
	return a;
}

// method 2
//这里还是利用了hash的思想，我利用python做这道题也是这样做的。
var twoSum = function(nums, target){
	var hash = {},
        result = [];
    for(var i = 0; i < nums.length; i++){
        hash[nums[i]] = i; 
    }
    for(var j = 0; j< nums.length; j++){
        var a = target - nums[j];
        if(hash[a]){
            result.push(j, hash[a]);
            break;
        }
    }
    return result;
}

File: leetcode-algorithms/001. Two Sum/two_sum.py
#!/usr/bin/env python
# Created by Bruce yuan on 18-1-23.


class Solution(object):
    def twoSum(self, nums, target):
        i = 0
        q = 0
        for i in range(0, len(nums)):
            m = target - nums[i]
            if m in nums:
                q = nums.index(m)
                if q != i:
                    break
        return [i, q]


class Solution2(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = {}
        for index, item in enumerate(nums):
            if d.get(target - item, None) or d.get(target - item, None) == 0:
                return [d.get(target - item), index]
            else:
                d[item] = index

